src/mbgl/algorithm/update_renderables.hpp

```diff
@@ -14,12 +14,14 @@ template <typename GetTileFn,
           typename CreateTileFn,
           typename RetainTileFn,
           typename RenderTileFn,
-          typename IdealTileIDs>
+          typename IdealTileIDs,
+          typename PrefetchedTileMap>
```

Comment:
I don't think `IdealTileIDs` nor `PrefetchedTileMap` need to be template parameters.

Can you change them to `const std::vector<OverscaledTileID>&` and `const std::map<OverscaledTileID, std::unique_ptr<Tile>> &`?

Reply:
Tests are using `MockTestData` instead of `Tile`. So they need to be template parameters for the tests.

Reply:
Also `idealTiles` are defined as `set` in tests: `std::set<mbgl::OverscaledTileID>`

Reply:
Thanks.